### PR TITLE
V2 Enable rootless

### PR DESCRIPTION
**Pull request by jwhonce**
* Enable running podman V2 rootless
* Fixed cobra.PersistentPreRunE usage in all the commands
* Leveraged cobra.PersistentPreRunE/cobra.PersistentPostRunE to manage:
  * rootless
  * trace (--trace)
  * profiling (--cpu-profile)
  * initializing the registry copies of Image/Container engines
* Help and Usage templates autoset for all sub-commands

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

File: cmd/podmanV2/Makefile [hunk 1]
```diff
@@ -1,7 +1,9 @@
 all: podman podman-remote
 
+BUILD_TAGS='ABISupport systemd varlink seccomp selinux exclude_graphdriver_devicemapper'
```
Comment (rhatdan) [inline cmd/podmanV2/Makefile:3]: When we move this into the main repo, these should be autogenerated.

File: cmd/podmanV2/registry/registry.go [hunk 2]
```diff
@@ -112,6 +79,7 @@ func NewContainerEngine(cmd *cobra.Command, args []string) (entities.ContainerEn
 }
 
 func SubCommandExists(cmd *cobra.Command, args []string) error {
+	fmt.Fprintf(os.Stderr, "In %s.SubCommandExists\n", cmd.Name())
```
Comment (rhatdan) [inline cmd/podmanV2/registry/registry.go:82]: Left over debug?

File: pkg/domain/infra/abi/system.go [hunk 3]
```diff
@@ -123,9 +123,10 @@ func (ic *ContainerEngine) SetupRootless(cmd *cobra.Command) error {
 		}
 	}
 
-	if !executeCommandInUserNS(cmd) {
-		return nil
-	}
+	// Check moved into presentation layer
```
Comment (rhatdan) [inline pkg/domain/infra/abi/system.go:126]: Why not delete?

File: cmd/podmanV2/Makefile (hunk 1 above)
Comment (jwhonce) [reply cmd/podmanV2/Makefile:3]: Correct

File: cmd/podmanV2/registry/registry.go (hunk 2 above)
Comment (jwhonce) [reply cmd/podmanV2/registry/registry.go:82]: Yup, I thought I had caught them all.

File: pkg/domain/infra/abi/system.go (hunk 3 above)
Comment (jwhonce) [reply pkg/domain/infra/abi/system.go:126]: I wanted to give a hint where it went. :-) I'll remove it with next push.

**Comment by baude** [comment]
assuming @rhatdan review was a LGTM from earlier

**Pull request merged**